interface/forms/vitals/C_FormVitals.class.php

```diff
@@ -83,40 +93,112 @@ public function default_action()
+        $records = $vitalsService->getVitalsHistoryForPatient($GLOBALS['pid'], $form_id);
+
+        foreach ($records as $result) {
             $results[$i]['id'] = $result['id'];
             $results[$i]['uuid'] = $result['uuid'];
-            $results[$i]['encdate'] = substr($result['encdate'], 0, 10);
+//            $results[$i]['encdate'] = substr($result['encdate'], 0, 10);
             $results[$i]['date'] = $result['date'];
             $results[$i]['activity'] = $result['activity'];
-            $results[$i]['bps'] = $result['bps'];
-            $results[$i]['bpd'] = $result['bpd'];
-            $results[$i]['weight'] = $result['weight'];
-            $results[$i]['height'] = $result['height'];
-            $results[$i]['temperature'] = $result['temperature'];
-            $results[$i]['temp_method'] = $result['temp_method'];
-            $results[$i]['pulse'] = $result['pulse'];
-            $results[$i]['respiration'] = $result['respiration'];
-            $results[$i]['BMI'] = $result['BMI'];
-            $results[$i]['BMI_status'] = $result['BMI_status'];
             $results[$i]['note'] = $result['note'];
-            $results[$i]['waist_circ'] = $result['waist_circ'];
-            $results[$i]['head_circ'] = $result['head_circ'];
-            $results[$i]['oxygen_saturation'] = $result['oxygen_saturation'];
-            $results[$i]['oxygen_flow_rate'] = $result['oxygen_flow_rate'];
-            $results[$i]['ped_weight_height'] = $result['ped_weight_height'];
-            $results[$i]['ped_bmi'] = $result['ped_bmi'];
-            $results[$i]['ped_head_circ'] = $result['ped_head_circ'];
+            foreach ($vitalsColumn as $column)
+            {
+                $results[$i][$column] = $result[$column];
+
+                if (isset($result[$column . '_interpretation_id']))
+                {
+                    $results[$i][$column . '_intrepration_id'] = $result[$column . '_interpretation_id'];
+                    $results[$i][$column . '_intrepration_code'] = $result[$column . '_interpretation_code'];
+                    $results[$i][$column . '_intrepration_text'] = $result[$column . '_interpretation_text'];
+                }
+            }
+            $this->convertVitalsToCorrectMeasurementUnit($results[$i], $result);
+
+            // weight has a unique lb/oz display so we will handle that here.
+            $results[$i]['weight']['usa'] = $vitals->display_weight($results[$i]['weight']['usa']);
             $i++;
         }
 
         $this->assign("vitals", $vitals);
         $this->assign("results", ($results ?? null));
 
+        $this->assign('interpretation_options', $this->get_interpretation_list_options());
+
         $this->assign("VIEW", true);
         return $this->fetch($this->template_dir . $this->template_mod . "_new.html");
     }
 
+    private function get_interpretation_list_options()
+    {
+        $listService = new \OpenEMR\Services\ListService();
```

Comment:
would put in `use` clause at top

Reply:
Thanks, I plan on doing that.

Reply:
guessing this change not yet pushed to the PR

Reply:
Yea, overlooked this one.  I'm updating it now.  Dealing with some wierd database issues testing against a fresh database install.  Didn't happen until I rebased on top of master.